waspc/packages/wasp-config/src/mapUserSpecToAppSpecDecls.ts

```diff
@@ -64,60 +71,39 @@ export function mapUserSpecToDecls(
       emailSender,
       websocket
     ),
-  })
-
-  for (const [actionName, actionConfig] of actions.entries()) {
-    decls.push({
-      declType: 'Action',
-      declName: actionName,
-      declValue: mapOperationConfig(actionConfig, parseEntityRef),
-    })
-  }
-
-  for (const [queryName, queryConfig] of queries.entries()) {
-    decls.push({
-      declType: 'Query',
-      declName: queryName,
-      declValue: mapOperationConfig(queryConfig, parseEntityRef),
-    })
   }
 
-  for (const [apiName, apiConfig] of apis.entries()) {
-    decls.push({
-      declType: 'Api',
-      declName: apiName,
-      declValue: mapApiConfig(apiConfig, parseEntityRef),
-    })
-  }
-
-  for (const [jobName, jobConfig] of jobs.entries()) {
-    decls.push({
-      declType: 'Job',
-      declName: jobName,
-      declValue: mapJob(jobConfig, parseEntityRef),
-    })
-  }
-
-  for (const [
-    apiNamespaceName,
-    apiNamespaceConfig,
-  ] of apiNamespaces.entries()) {
-    decls.push({
-      declType: 'ApiNamespace',
-      declName: apiNamespaceName,
-      declValue: mapApiNamespace(apiNamespaceConfig),
-    })
-  }
+  return makeDeclsArray({
+    App: [appDecl],
+    Page: pageDecls,
+    Route: routeDecls,
+    Action: actionDecls,
+    Query: queryDecls,
+    Api: apiDecls,
+    Job: jobDecls,
+    ApiNamespace: apiNamespaceDecls,
+    Crud: crudDecls,
+  })
+}
 
-  for (const [crudName, crudConfig] of spec.cruds.entries()) {
-    decls.push({
-      declType: 'Crud',
-      declName: crudName,
-      declValue: mapCrud(crudConfig, parseEntityRef),
-    })
-  }
+function makeDeclsArray(decls: {
+  [Type in AppSpec.Decl['declType']]: AppSpec.GetDeclForType<Type>[]
+}): AppSpec.Decl[] {
+  return Object.values(decls).flatMap((decl) => [...decl])
+}
 
-  return decls
+function mapToDecls<T, const DeclType extends AppSpec.Decl['declType']>(
+  configs: Map<string, T>,
+  type: DeclType,
+  configToDeclValue: (
+    config: T
+  ) => AppSpec.GetDeclForType<DeclType>['declValue']
+) {
```

Comment:
I can't specify a return type of this function because of some weird error, but I stopped digging into it.

If you're curious, here's the [playground](https://www.typescriptlang.org/play/?#code/FAUwHgDg9gTgLgAjgTwiBBxEcCiA7OGZAMVgGkBLPAEwB5KaFw4QaBnBfQ5AbQHIA1lWp8AugD4EAXk5hCAQwDGcWsAScCRADRqEAbwRCaALgQNqCAL7BxwUJFiIUaDd2m6APvsPDTfNoRUAOZ8ANwIAG7yADYAriCmATDBVp7eRtR+eLEAtgBGIDBhkTHxptn5hal2AGaxeMoUUHgI1FAAylA52AAWwfTCTHKs1BxcRPwZYuIAFBmm5lolcQmY2OMk5MIDNOL8UStiAJSmWLiamzDmO9SSerow2LEwLffq6hk678vxutaWQA).